systick.c

```diff
@@ -0,0 +1,100 @@
+/*
+  Not part of Grbl. KeyMe specific.
+
+  SysTick implementation. Global SysTick value is updated.
+  Callbacks can be registed to be called after a certain time.
+
+  SysTick uses Timer1. Timer1 should not be used anywhere else.
+
+*/
+#include "systick.h"
+#include "nuts_bolts.h"
+
+#define MAX_CALLBACKS 32
+
+callback_t systick_callbacks_array[MAX_CALLBACKS];
+uint8_t systick_len;  // Length of callback array 
+
+void systick_service_callbacks()
+{
+  if (!systick_len) {
+    return;
+  }
+  // Assumes the callback array is sorted 
+  // according to lowest callback time
+  if (systick_callbacks_array[0].callback_time <= SysTick) {
```

Comment:
My reasoning with not implementing a loop is that I do not want to call multiple callbacks in one execution of systick_service_callback. I would rather have the next callback being called the next time the protocol loop rolls over. If the callbacks take very long and there are multiple callbacks at the same time, the callbacks will hold back the rest of the system. I would rather have it as a softer real time. Thoughts?